packages/browser/src/profiling/hubextensions.ts

```diff
@@ -86,21 +71,52 @@ function wrapTransactionWithProfiling(transaction: Transaction): Transaction {
 
   const client = getCurrentHub().getClient();
   const options = client && client.getOptions();
+  if (!options) {
+    __DEBUG_BUILD__ && logger.log('[Profiling] Profiling disabled, no options found.');
+    return transaction;
+  }
 
-  // @ts-ignore not part of the browser options yet
-  const profilesSampleRate = (options && options.profilesSampleRate) || 0;
-  if (profilesSampleRate === undefined) {
-    if (__DEBUG_BUILD__) {
-      logger.log('[Profiling] Profiling disabled, enable it by setting `profilesSampleRate` option to SDK init call.');
-    }
+  // @ts-ignore profilesSampler is not part of the browser options yet
+  const profilesSampler = options.profilesSampler;
+  // @ts-ignore profilesSampleRate is not part of the browser options yet
+  let profilesSampleRate: number | boolean | undefined = options.profilesSampleRate;
+
+  // Prefer sampler to sample rate if both are provided.
+  if (typeof profilesSampler === 'function') {
+    profilesSampleRate = profilesSampler({ transactionContext: transaction.toContext() });
```

Comment:
Yeah, I will need to fix that later, but it's something I'm aware of. Afaik customSamplingContext is not stored anywhere (or I couldn't find it) so because we wrap the transaction after it's already been created, we cannot access it anymore.

Reply:
If that's the case, thoughts on not supporting the `profilesSampler` for now and resolving this later rather than a temporary implementation and then breaking compatibility in the future?

Reply:
Good call, removing it